### PR TITLE
refactor: enhance name service

### PR DESCRIPTION
**Pull request by guanbinrui**
## Description

+ Add test example in Debugger console.
+ Enhance cache hitting algorithm.
+ A new class `NameServiceResolver` for creating a resolver.

Closes mf-1136

## Type of change

- [ ] Documentation
- [x] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

### CODE REVIEW

File: packages/plugins/Debugger/src/SNSAdaptor/components/ConsoleContent.tsx [hunk 1]
```diff
@@ -86,16 +90,33 @@ export function ConsoleContent(props: ConsoleContentProps) {
...
-                <Button
-                    size="small"
-                    onClick={async () => {
-                        const token = await onSelectFungibleToken()
-                        console.log(token)
-                    }}>
-                    Select Fungible Token
-                </Button>
+                <Typography variant="body2">
+                    <span>{lookedAddress}</span>
```
Comment (UncleBill) [inline packages/plugins/Debugger/src/SNSAdaptor/components/ConsoleContent.tsx:95]: Add a `<input />` to input custom address.

File: packages/plugins/Debugger/src/SNSAdaptor/components/ConsoleContent.tsx [hunk 2]
```diff
@@ -86,16 +90,33 @@ export function ConsoleContent(props: ConsoleContentProps) {
...
+                        Lookup Address
+                    </Button>
+                </Typography>
+            ),
+        },
+        {
+            name: 'Reversed Name',
+            content: (
+                <Typography variant="body2">
+                    <span>{reversedName}</span>
```
Comment (UncleBill) [inline packages/plugins/Debugger/src/SNSAdaptor/components/ConsoleContent.tsx:110]: Add a `<input />` for custom ens.

File: packages/plugins/Debugger/src/SNSAdaptor/components/ConsoleContent.tsx (hunk 1 above)
Comment (guanbinrui) [reply packages/plugins/Debugger/src/SNSAdaptor/components/ConsoleContent.tsx:95]: It's a read-only field for test purposes.

File: packages/plugins/Debugger/src/SNSAdaptor/components/ConsoleContent.tsx (hunk 2 above)
Comment (guanbinrui) [reply packages/plugins/Debugger/src/SNSAdaptor/components/ConsoleContent.tsx:110]: It's a read-only field for test purposes.

File: packages/plugins/EVM/src/state/NameService/ENS.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        }
+    }
+
+    async reverse(chainId: ChainId, address: string) {
+        if (chainId !== ChainId.Mainnet) return
+
+        try {
+            const ens = await this.createENS()
+            const name = await ens.reverse(address)
+            if (!name.endsWith('.eth')) return
```
Comment (guanbinrui) [inline packages/plugins/EVM/src/state/NameService/ENS.ts:38]: CC @UncleBill, Did I miss your near updates here?

**Pull request merged**